src/main/java/gndata/lib/srv/ResourceAdapter.java

```diff
@@ -77,4 +79,39 @@ public boolean equals(Object o) {
     public int hashCode() {
         return resource.hashCode();
     }
+
+
+    /* resource action API: TODO refactor in a separate interface later */
+
+
+    public Optional<String> getLabel() {
+        Statement labelSt = resource.getProperty(RDFS.label);
+        return labelSt == null ? Optional.ofNullable(null) : Optional.of(labelSt.getObject().asLiteral().getString());
+    }
+
+
+    public Resource addLiteral(Property p, String data, RDFDatatype dtype) {
+        Literal o = ResourceFactory.createTypedLiteral(data, dtype);
+        return resource.addLiteral(p, o);
+    }
+
+    public void removeObjectProperty(Property p, Resource obj) {
```

Comment:
- does not work because literals have to be filtered out first
- the method should actually be provided with a parent and a child resource and remove all object properties that they share